pkg/scaling/executor/scale_executor.go

```diff
@@ -86,47 +87,35 @@ func (e *scaleExecutor) updateLastActiveTime(ctx context.Context, logger logr.Lo
 	return err
 }
 
-func (e *scaleExecutor) setCondition(ctx context.Context, logger logr.Logger, object interface{}, status metav1.ConditionStatus, reason string, message string, setCondition func(kedav1alpha1.Conditions, metav1.ConditionStatus, string, string)) error {
-	var patch runtimeclient.Patch
-
-	runtimeObj := object.(runtimeclient.Object)
-	switch obj := runtimeObj.(type) {
-	case *kedav1alpha1.ScaledObject:
-		patch = runtimeclient.MergeFrom(obj.DeepCopy())
-		setCondition(obj.Status.Conditions, status, reason, message)
-	case *kedav1alpha1.ScaledJob:
-		patch = runtimeclient.MergeFrom(obj.DeepCopy())
-		setCondition(obj.Status.Conditions, status, reason, message)
-	default:
-		err := fmt.Errorf("unknown scalable object type %v", obj)
-		logger.Error(err, "Failed to patch Objects Status")
-		return err
-	}
-
-	err := e.client.Status().Patch(ctx, runtimeObj, patch)
-	if err != nil {
-		logger.Error(err, "Failed to patch Objects Status")
-	}
-	return err
-}
-
-func (e *scaleExecutor) setReadyCondition(ctx context.Context, logger logr.Logger, object interface{}, status metav1.ConditionStatus, reason string, message string) error {
-	active := func(conditions kedav1alpha1.Conditions, status metav1.ConditionStatus, reason string, message string) {
-		conditions.SetReadyCondition(status, reason, message)
-	}
-	return e.setCondition(ctx, logger, object, status, reason, message, active)
+func (e *scaleExecutor) setReadyCondition(ctx context.Context, object interface{}, status metav1.ConditionStatus, reason string, message string) error {
+	return util.SetStatusConditions(ctx, e.client, object, &kedav1alpha1.Conditions{
```

Comment:
I think this call will replace all existing conditions of another type that are already defined on the ScaledObject's Status. See how Conditions are being set:
https://github.com/kedacore/keda/blob/98509a83a15dd4a8dcaacf1aecba99fc38162c70/apis/keda/v1alpha1/condition_types.go#L126-L191

Reply:
Ah that's a good catch - I put in a first idea that seems to pass that one failing test, I'm currently only on my chromebook that can't run the full test suit without blowing up.
I haven't really thought about the implementation too much on that one and it's copying essentially the same switch statement for the two types there are.

One question for this I suppose is how much does performance matter on this code path? Do we need to worry about allocs or speed necessarily here?

Reply:
Sorry for the late reply, I have been really busy, Kubecon and other stuff.

We should really care about performance on this. Status might be updated quite often and there might be tons of ScaledObjects to update.

Reply:
I think this will definitely be a bit more involved with that in mind :thinking: the last implementation I wrote broke something else, I'm feeling the ways the two functions are originally designed are subtly different so I'll have to spend more time understanding - pretty busy myself currently so have not made any significant progress at this point.

Reply:
Sure, let's keep this open :)